build-system/tasks/e2e/index.js

```diff
@@ -66,6 +67,7 @@ function cleanUp_() {
   if (webServerProcess_ && !webServerProcess_.killed) {
     webServerProcess_.kill('SIGINT');
   }
+  exitCtrlcHandler(webServerProcess_.pid);
```

Comment:
This will only work if you've first called `createCtrlcHandler`. (It creates a child process that sits around waiting for a Ctrl+C signal when the main process is too busy to listen for keyboard input.)

Reply:
Sorry, I should have been more clear in my description. The actual e2e function exits cleanly. It's `gulp serve` that causes this task to hang, because it runs `createCtrlcHandler`.

I'm confused why we shouldn't return promises? That article linked above doesn't say not to...? It's also included in the gulp 4 [docs](https://gulpjs.com/docs/en/getting-started/async-completion#returning-a-promise)?

The mocha runner provides a callback function so I thought it was cleanest to resolve the promise there instead of subscribing to events.

Reply:
I see what you're saying.

1) The function `exitCtrlcHandler` happens to work here because you're passing it the process ID of `gulp serve`, and not the process ID of the Ctrl+C handler started by `gulp serve`. Since this could leave behind an orphaned process, could you try calling `kill()` with `SIGKILL` on the webserver process so it cleanly exits, similar to how visual diff does it below?

https://github.com/ampproject/amphtml/blob/68bf94bd43986d107a22d28fa2bcff07add32177/build-system/tasks/visual-diff/index.js#L733-L743

2) Scratch what I said about the `mocha.run()` changes for this PR. If the above fix works, let's merge this PR with just that change.